zimui/src/components/TopicCard.vue

```diff
@@ -0,0 +1,152 @@
+<script setup lang="ts">
+import { ref } from 'vue'
+
+import imageData from '../assets/card_default_bg.png'
+
+const image = ref(imageData)
+
+defineProps({
+  data: {
```

Comment:
I dislike this because we have no contract about what should be in the props. It's very error prone